File: week1/04-cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int change;
    int coins = 0;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0);

    // 25c
    if (change >= 25)
    {
        coins += change / 25;
        change = change % 25;
    }

    // 10c
    if (change >= 10)
    {
        coins += change / 10;
        change = change % 10;
    }

    // 5c
    if (change >= 5)
    {
        coins += change / 5;
        change = change % 5;
    }

    // 1c
    if (change >= 1)
    {
        coins += change / 1;
    }

    // Print coins used
    printf("%d\n", coins);
}

File: week6/04-sentimental-cash/cash.py
from cs50 import get_float


def main():
    cash()


def cash():
    while (True):
        print(9)
        change = get_float("Change: ")
        if change > 0:
            break
    coins = 0

    # 25c
    if change >= 0.25:
        coins += change // 0.25
        change = round(change % 0.25, 2)
        # print(25, coins, change)

    # 10c
    if change >= 0.10:
        coins += change // 0.10
        change = round(change % 0.10, 2)
        # print(10, coins, change)

    # 5c
    if change >= 0.05:
        coins += change // 0.05
        change = round(change % 0.05, 2)
        # print(5, coins, change)

    # 1c
    if change >= 0.01:
        coins += change // 0.01
        change = round(change % 0.01, 2)
        # print(1, coins, change)

    print(f"{int(coins)}")


main()
